src/js/core/player.js

```diff
@@ -792,17 +792,53 @@ function name(): {country: string, firstName: string, lastName: string} {
     const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
     const firstNameRow = playerNames.first[country].find(row => row[1] >= fnRand);
     if (firstNameRow === undefined) {
-        throw new Error(`Undefined firstNameRow (fnRand=${fnRand}`);
+        throw new Error(`Undefined firstNameRow (fnRand=${fnRand})`);
+    }
+    let firstName = firstNameRow[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 50% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 50% will use it fully
+        //   e.g. {Xavier Yanick Zed}
+        const abbrev = random.uniform(0, 1);
+        if (abbrev < 0.5 || country === "USA") {
+            // initialize it
+            middleName = `${middleName.charAt(0)}.`;
+        } else {
+            // use it fully
```

Comment:
Ok. I kept it just in case you had alternative ideas, such as wrapping a middle-name in quotes to make a "nickname", e.g. "William 'Billy' Smith". I can remove it easily.